tests/capi/unittest_capi_service_extension.cc

```diff
@@ -390,6 +390,113 @@ _extension_test_imgclf (ml_service_h handle, gboolean is_pipeline)
   _free_test_data (tdata);
 }
 
+/**
+ * @brief Callback function for scenario test.
+ */
+static void
+_extension_test_llamacpp_cb (
+    ml_service_event_e event, ml_information_h event_data, void *user_data)
+{
+  extension_test_data_s *tdata = (extension_test_data_s *) user_data;
+  ml_tensors_data_h data = NULL;
+  void *_raw = NULL;
+  size_t _size = 0;
+  int status;
+
+  switch (event) {
+    case ML_SERVICE_EVENT_NEW_DATA:
+      ASSERT_TRUE (event_data != NULL);
+
+      status = ml_information_get (event_data, "data", &data);
+      EXPECT_EQ (status, ML_ERROR_NONE);
+
+      status = ml_tensors_data_get_tensor_data (data, 0U, &_raw, &_size);
+      EXPECT_EQ (status, ML_ERROR_NONE);
+
+      g_print ("%s", (char *) _raw);
+
+      if (tdata)
+        tdata->received++;
+      break;
+    default:
+      break;
+  }
+}
+
+/**
+ * @brief Internal function to run test with ml-service extension handle.
+ */
+static inline void
+_extension_test_llamacpp (ml_service_h handle, gboolean is_pipeline)
+{
+  extension_test_data_s *tdata;
+  ml_tensors_info_h info;
+  ml_tensors_data_h input;
+  int status;
+  gsize len = 0;
+  gchar *contents = NULL;
+
+  g_autofree gchar *data_file = _get_data_path ("input.txt");
```

Comment:
I suggest removing input.txt file, and get predefined input-text.
```
const gchar input_text[] = "Hello my name is";
...
ml_tensors_data_set_tensor_data (input, 0U, g_strdup (input_text), strlen (input_text));
```

Reply:
Thank you for taking the time to review.
I've updated it. Thank you.